src/ProjectNeon/Assets/Features/Effects/ForNumberOfTurns.cs

```diff
@@ -1,38 +1,32 @@
 ﻿
 /**
  * Wraps the execution of an effect so it is only executed for a certain amount of turns.
+ * 
+ * Turn ending decrease the remainingDuration in one. 
  */
-public class ForNumberOfTurns : ITemporalState, Effect
+public class ForNumberOfTurns : Effect
 {
     
     private int _remainingDuration;
 
     private Effect _effect;
 
-    private bool _isDebuff;
+    public bool IsActive => _remainingDuration >= 0;
 
-    public IStats Stats { get; } = new StatAddends();
-    public bool IsDebuff => _isDebuff;
```

Comment:
How will poison get cleansed if Temporal State is no longer implemented? 

This class still needs to implement Temporal State.

Reply:
@SilasReinagel Poison isn't implemented yet; I though in implementing it as an effect that causes damage over time.

Reply:
We can't take away the IsDebuff flag, since that breaks the existing effect `CleanseDebuffs`

Reply:
I still don't understand why we stopped implementing `ITemporalState`. There seems to be no reason for it.

I will merge this and open a bug ticket.